src/App.js

```diff
@@ -1,10 +1,34 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import data from './data/dummy.json';
 
 const App = () => {
+  const [response, setResponse] = useState({});
+  const [score, setScore] = useState(0);
+  const [displayScore, setDisplayScore] = useState(false);
+
+  const handleResponse = e => {
+    const selectedAnswer = { [e.target.name]: e.target.value };
+    setResponse({ ...response, ...selectedAnswer });
+  };
+
+  const checkAnswers = e => {
+    e.preventDefault();
+    const handleScore = data
+      .map(question => {
+        const responseAnswerId = response[question.id];
+        const responseObj = question.answers.find(
+          answer => answer.id === Number(responseAnswerId),
+        );
+        return responseObj.isCorrect;
+      })
+      .filter(item => item).length;
```

Comment:
Here, we can use `reduce` since we are reducing our array from 5 items to a single number. Can you try changing `handleScore` to use `reduce` instead of `.map.filter`?

Here's a quick summary which helps me remember which function to use depending on the output I want:

- `map` — changes an array of 5 to an array of 5 (output length is the same as input length)
- `filter` — changes an array of 5 to an array of 0-5 (output length is usually less than the input length)
- `reduce` — changes an array of 5 to a single value of anything (output is anything: number, object, string, etc)